src/UI/Utils/useIsDirty.test.tsx

```diff
@@ -0,0 +1,175 @@
+    }
+  );
+  render(component);
+  expect(screen.queryByText("true")).not.toBeInTheDocument();
+  const button = screen.getByText("Change Data");
+
+  await act(async () => {
+    await userEvent.click(button);
+  });
+
+  expect(screen.queryByText("true")).not.toBeInTheDocument();
+});
+test("GIVEN useIsDirty WHEN hook's parameter that is being checked is changed from different than default THEN isDirty value is truthy and span is visible", async () => {
+  Field.nestedEditable, ServiceInstance.nestedEditable.candidate_attributes;
+  const { component } = setup(
+    {
+      property: "",
+    },
+    undefined,
+    {
+      property: "changed",
+    }
+  );
+  render(component);
+  const button = screen.getByText("Change Data");
+
+  await act(async () => {
+    await userEvent.click(button);
+  });
+
+  expect(screen.getByText("true")).toBeVisible();
+});
+
+test("GIVEN useIsDirty WHEN hook's parameter that is being checked is changed from empty value to the same that hook's baseObject has THEN isDirty value is falsy and span isn't visible", async () => {
+  Field.nestedEditable, ServiceInstance.nestedEditable.candidate_attributes;
+  const { component } = setup(
+    {
+      property: "",
+    },
+    {
+      property: "baseValue",
+    },
+    {
+      property: "baseValue",
+    }
+  );
+  render(component);
+  expect(screen.getByText("true")).toBeVisible();
+
+  const button = screen.getByText("Change Data");
+
+  await act(async () => {
+    await userEvent.click(button);
+  });
+
+  expect(screen.queryByText("true")).not.toBeInTheDocument();
+});
+test("GIVEN useIsDirty WHEN hook's parameter that is being checked is changed from one different value to another different than hook's baseObject THEN isDirty value is truthy and span always visible", async () => {
+  Field.nestedEditable, ServiceInstance.nestedEditable.candidate_attributes;
+  const { component } = setup(
+    {
+      property: "",
+    },
+    {
+      property: "baseValue",
+    },
+    {
+      property: "changed",
+    }
+  );
+  render(component);
+  expect(screen.getByText("true")).toBeVisible();
+
+  const button = screen.getByText("Change Data");
+
+  await act(async () => {
+    await userEvent.click(button);
+  });
+
+  expect(screen.getByText("true")).toBeVisible();
+});
+
+test("GIVEN useIsDirty WHEN hook's parameter that is being checked is changed to be different than hook's baseObject THEN isDirty value is truthy and span is visible", async () => {
+  Field.nestedEditable, ServiceInstance.nestedEditable.candidate_attributes;
+  const { component } = setup(
+    {
+      property: "baseValue",
+    },
+    {
+      property: "baseValue",
+    },
+    {
+      property: "changed",
+    }
+  );
+  render(component);
+  expect(screen.queryByText("true")).not.toBeInTheDocument();
+
+  const button = screen.getByText("Change Data");
+
+  await act(async () => {
+    await userEvent.click(button);
+  });
+
+  expect(screen.getByText("true")).toBeVisible();
```

Comment:
Aside from the last test case, it might be nice to have a loop, cause all other tests are a repetition, the input of the setup method is the only thing that changes